server/src/main/java/io/deephaven/server/table/stats/ObjectChunkedStats.java

```diff
@@ -0,0 +1,117 @@
+    }
+
+    @Override
+    public Table processChunks(final RowSet rowSet, final ColumnSource<?> columnSource, boolean usePrev) {
+        long count = 0;
+        int uniqueCount = 0;
+
+        final TObjectLongHashMap<Object> countValues = new TObjectLongHashMap<>();
+        boolean useSet = false;
+        final Set<Object> uniqueValues = new HashSet<>();
+
+        try (final ChunkSource.GetContext getContext = columnSource.makeGetContext(CHUNK_SIZE)) {
+            final RowSequence.Iterator rsIt = rowSet.getRowSequenceIterator();
+
+            while (rsIt.hasMore()) {
+                // Grab up to the next CHUNK_SIZE rows
+                final RowSequence nextKeys = rsIt.getNextRowSequenceWithLength(CHUNK_SIZE);
+
+                final ObjectChunk<?, ? extends Values> chunk =
+                        (usePrev ? columnSource.getPrevChunk(getContext, nextKeys)
+                                : columnSource.getChunk(getContext, nextKeys)).asObjectChunk();
+                final int chunkSize = chunk.size();
+                for (int ii = 0; ii < chunkSize; ii++) {
+                    final Object val = chunk.get(ii);
+
+                    if (val == null) {
+                        continue;
+                    }
+
+                    count++;
+
+                    if (useSet) {
+                        uniqueValues.add(val);
+                    } else if (uniqueCount > maxUniqueToCollect) {
+                        // we no longer need to track counts for these items; fall back to a Set to get at least a count
+                        uniqueValues.addAll(countValues.keySet());
+                        countValues.clear();
+                        uniqueValues.add(val);
+                        useSet = true;
+                    } else if (countValues.adjustOrPutValue(val, 1, 1) == 1) {
+                        uniqueCount++;
+                    }
+                }
+            }
+
+            final int numUnique;
+            String[] uniqueKeys;
+            long[] uniqueCounts;
+            if (useSet) {
+                numUnique = uniqueValues.size();
+                uniqueKeys = CollectionUtil.ZERO_LENGTH_STRING_ARRAY;
+                uniqueCounts = CollectionUtil.ZERO_LENGTH_LONG_ARRAY;
+            } else {
+                numUnique = countValues.size();
+                List<Map.Entry<String, Long>> sorted = new ArrayList<>();
+
+                countValues.forEachEntry((o, c) -> {
+                    sorted.add(Map.entry(Objects.toString(o), c));
+                    return true;
+                });
+                sorted.sort(Map.Entry.comparingByValue());
+
+                int resultCount = Math.min(maxUniqueToDisplay, sorted.size());
+                uniqueKeys = new String[resultCount];
+                uniqueCounts = new long[resultCount];
+                Iterator<Map.Entry<String, Long>> iter = sorted.iterator();
+                for (int i = 0; i < resultCount && iter.hasNext(); i++) {
```

Comment:
If you're short-circuiting on `!iter.hasNext()`, something is wrong...

Reply:
That's true - this was a late addition and I thought I was adding extra safety, but you're right that it doesn't make sense to exist.